### PR TITLE
Weighted algorithm

### PR DESCRIPTION
**Pull request by Thunae**
This branch implements the weighted algorithm that our application uses to choose a restaurant. This weighs restaurants with a price level closer to the user's request as more favorable. 
In addition, the extra search parameters are added to the restaurant search to provide more variety.
Finally, the restaurant choosing mechanism is added to its own class and integrated into the existing test.

### CODE REVIEW

File: src/main/java/com/google/sps/data/Restaurant.java [hunk 1]
```diff
@@ -25,13 +25,16 @@
     private final double rating;
     @SerializedName(value = "businessStatus", alternate = "business_status")
     private final String businessStatus;
+    @SerializedName(value = "priceLevel", alternate = "price_level")
+    private final int priceLevel;
     private final Photo[] photos;
 
-    public Restaurant(String name, double rating, String businessStatus, Photo[] photos) {
+    public Restaurant(String name, double rating, String businessStatus, int priceLevel, Photo[] photos) {
```
Comment (lcatherine) [inline src/main/java/com/google/sps/data/Restaurant.java]: nit: why priceLevel and not price? 
jw since you have getPrice() below

**Comment by lyze** [comment]
It looks like we need to rebase this PR.

File: src/test/java/com/google/sps/ResponseTest.java [hunk 2]
```diff
@@ -35,13 +37,13 @@ public void NoRerolls() {
...
+    List<Restaurant> restArr = new ArrayList<>();
+    restArr.add(new Restaurant("McDonald's", 4.5, "OPERATIONAL", 2, new Photo[]{}));
+    Response response = new Response("OK", restArr);
+    RestaurantChooser.chooseRestaurant(response, 2);
 
     String actual = response.getStatus();
-    String expected = "NO_REROLLS";
-    assertEquals(expected, actual);
+    String expected = "ZERO_RESULTS";
+    Assert.assertEquals(expected, actual);
```
Comment (lcatherine) [inline src/test/java/com/google/sps/ResponseTest.java]: I think the assertEquals was imported so you don't need to add Assert

File: src/main/java/com/google/sps/data/Restaurant.java (hunk 1 above)
Comment (Thunae) [reply src/main/java/com/google/sps/data/Restaurant.java]: changed to price

File: src/test/java/com/google/sps/ResponseTest.java (hunk 2 above)
Comment (Thunae) [reply src/test/java/com/google/sps/ResponseTest.java]: done

File: src/main/java/com/google/sps/data/RestaurantChooser.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+import java.util.HashMap;
+import java.util.Random;
+
+public final class RestaurantChooser {
+    public static void chooseRestaurant(Response response, int requestedPrice){
+        if(response.getResults().size() == 0){
+            response.setStatus("ZERO_RESULTS");
+            return;
+        }
+        HashMap<String, Integer> restaurantScores = new HashMap<>();
```
Comment (lcatherine) [inline src/main/java/com/google/sps/data/RestaurantChooser.java:32]: Maybe I am not understanding this algorithm correctly, but why not

Map<Integer, String> ( scores : Restaurant )
then randomly select an index from the size of the keySet
then output the associated value?

**Pull request closed without merging**